mods/classic-start-menu-folders-icon.wh.cpp

```diff
@@ -0,0 +1,212 @@
+                                                         void** ppv) {
+    long result = CFSFolder__CreateFileFolderDefExtIcon_orig(
+        pThis, pItemID, pIDFOLDER, pBind, guid, ppv);
+
+    ICachedPrivateProfile* pICachedPrivateProfile;
+    LPWSTR folderString[264];
+    UINT csidl = CFSFolder__GetCSIDL(pThis);
+
+    if (!CheckCSIDL(csidl)) {
+        return result;
+    }
+
+    if (CFSFolder__GetDesktopIniForItem(pThis, pItemID,
+                                        &pICachedPrivateProfile) >= 0) {
+        if (GetFolderString(pICachedPrivateProfile, L".ShellClassInfo",
+                            L"IconResource", folderString, MAX_PATH) ||
+            GetFolderString(pICachedPrivateProfile, L".ShellClassInfo",
+                            L"IconFile", folderString, MAX_PATH)) {
+            return result;
+        }
+    }
+
+    IDefaultExtractIconInit* pdxi = NULL;
+    SHCreateDefaultExtractIcon(IID_PPV_ARGS(&pdxi));
+    pdxi->SetNormalIcon(NULL, SIID_PROG);
+    pdxi->SetOpenIcon(NULL, SIID_PROG);
+
+    pdxi->QueryInterface((IID)*guid, ppv);
+
+    if (pdxi) {
+        pdxi->Release();
+    }
+
+    return result;
+}
+
+long THISCALL (*CFSFolder_GetIconOf_orig)(CFSFolder*,
+                                          ITEMIDLIST_RELATIVE*,
+                                          unsigned int,
+                                          int*);
+long THISCALL CFSFolder_GetIconOf_hook(CFSFolder* pThis,
+                                       ITEMIDLIST_RELATIVE* pItemID,
+                                       unsigned int a3,
+                                       int* a4) {
+    long result = CFSFolder_GetIconOf_orig(pThis, pItemID, a3, a4);
+
+    ICachedPrivateProfile* pICachedPrivateProfile;
+    LPWSTR folderString[264];
+    IShellFolder* cfsFolder = (IShellFolder*)((CFSFolder*)pThis - 56);
+    UINT csidl = CFSFolder__GetCSIDL((CFSFolder*)cfsFolder);
+    IDFOLDER* idFolder =
+        CFSFolder__IsValidID((CFSFolder*)cfsFolder, pItemID, NULL);
+
+    if (!CheckCSIDL(csidl)) {
+        return result;
+    }
+
+    if (!idFolder) {
+        return result;
+    }
+
+    if (!IsFileFolder(idFolder)) {
+        return result;
+    }
+
+    if (CFSFolder__GetDesktopIniForItem((void*)cfsFolder, pItemID,
+                                        &pICachedPrivateProfile) <= 0) {
+        if (pICachedPrivateProfile) {
+            if (GetFolderString(pICachedPrivateProfile, L".ShellClassInfo",
+                                L"IconResource", folderString, MAX_PATH) ||
+                GetFolderString(pICachedPrivateProfile, L".ShellClassInfo",
+                                L"IconFile", folderString, MAX_PATH)) {
+                return result;
+            }
+            *a4 = Shell_GetStockImageIndex(SIID_PROG);
+        }
+    }
+
+    return result;
+}
+
+BOOL Wh_ModInit() {
+    Wh_Log(L"Init");
+
+    const WindhawkUtils::SYMBOL_HOOK windowsStorageDllHooks[] = {
```

Comment:
`windows.storage.dll` can't be represented by a variable name. Please use a comment instead, for example:
```suggestion
    // windows.storage.dll
    const WindhawkUtils::SYMBOL_HOOK windowsStorageHooks[] = {
```